### DISCUSSION TITLE
Running batches of objective function to reduce the number of trials.

### DISCUSSION THREAD

**Discussion by alfsachin** [optuna/optuna#5132] [Q&A] [answered]
Hi Optuna Community,

I'm working on a project where I need to run multiple parallel backtests based on Optuna's suggestions, and then use the outcomes for further suggestions. For instance, I'd like to run 100 backtests in one batch (of a trial).  My goal is to reduce the serial run time of the entire process. 

The process I envision is:

1. Start an Optuna study.
2. Generate a batch of parameter suggestions (e.g., 100).
3. Execute parallel backtests using these parameters.
4. Feed the backtest results back into Optuna (to learn from them)
5. Use this information for the next batch of suggestions.

I'm seeking insights on how to run many objective functions in batch, rather in serial mode.  Any guidance or shared experiences would be invaluable!

Thanks,

Sachin

**Comment by nzw0301** [comment, marked as answer]
Hello, ask-and-tell interface allows us to perform bath optimisation, see, 
https://optuna.readthedocs.io/en/stable/tutorial/20_recipes/009_ask_and_tell.html#batch-optimization for the concrete example. Does this work for you?

**Reply by alfsachin** [reply]
Thankyou, this is what I was looking for.  

Would you happen to know which samplers perform best with batch sampling and fewer trials?
